File: client/lib/models/responses/macros.dart
/// Macro bundle — used as totals, target, and remaining on the day
/// payload and per-entry / per-item macros. Fields stay null when the
/// source food doesn't carry them (fiber, sodium), so the client renders
/// a dash rather than a misleading zero.
class Macros {
  Macros({
    this.kcal,
    this.proteinG,
    this.carbG,
    this.fatG,
    this.fiberG,
    this.sodiumMg,
  });

  final double? kcal;
  final double? proteinG;
  final double? carbG;
  final double? fatG;
  final double? fiberG;
  final double? sodiumMg;

  factory Macros.fromJson(Map<String, dynamic> json) {
    return Macros(
      kcal: (json['kcal'] as num?)?.toDouble(),
      proteinG: (json['proteinG'] as num?)?.toDouble(),
      carbG: (json['carbG'] as num?)?.toDouble(),
      fatG: (json['fatG'] as num?)?.toDouble(),
      fiberG: (json['fiberG'] as num?)?.toDouble(),
      sodiumMg: (json['sodiumMg'] as num?)?.toDouble(),
    );
  }
}

File: api/src/main/java/com/example/athletehub/dto/Macros.java
package com.example.athletehub.dto;

import java.math.BigDecimal;

/**
 * Macro bundle — used as the totals, target, and remaining values on the
 * day endpoint and the per-entry payload. Fiber and sodium stay null when
 * the underlying food rows don't carry them, so the client can render a
 * dash rather than a misleading zero.
 */
public record Macros(
        BigDecimal kcal,
        BigDecimal proteinG,
        BigDecimal carbG,
        BigDecimal fatG,
        BigDecimal fiberG,
        BigDecimal sodiumMg
) {
    public static Macros zero() {
        return new Macros(
                BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO,
                BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO);
    }
}
